### PR TITLE
Use Kokkos swap and introduce KokkosExt header for StdAlgorithms

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Review by dalg24 — CHANGES_REQUESTED** [review]
Not using Kokkos swap until it gets "promoted" to core

File: src/kokkos_ext/ArborX_DetailsKokkosExtStdAlgorithms.hpp [hunk 1]
```diff
@@ -9,27 +9,17 @@
  * SPDX-License-Identifier: BSD-3-Clause                                    *
  ****************************************************************************/
 
-#ifndef ARBORX_DETAILS_KOKKOS_EXT_SWAP_HPP
-#define ARBORX_DETAILS_KOKKOS_EXT_SWAP_HPP
+#ifndef ARBORX_DETAILS_KOKKOS_EXT_STD_ALGORITHMS_HPP
+#define ARBORX_DETAILS_KOKKOS_EXT_STD_ALGORITHMS_HPP
 
-#include <Kokkos_Macros.hpp>
+#include <Kokkos_StdAlgorithms.hpp>
```
Comment (dalg24) [inline src/kokkos_ext/ArborX_DetailsKokkosExtStdAlgorithms.hpp:15]: This is not desirable

**Comment by aprokop** [comment]
> Not using Kokkos swap until it gets "promoted" to core

Why not?

File: src/kokkos_ext/ArborX_DetailsKokkosExtStdAlgorithms.hpp (hunk 1 above)
Comment (aprokop) [reply src/kokkos_ext/ArborX_DetailsKokkosExtStdAlgorithms.hpp:15]: Please elaborate.

Comment (dalg24) [reply src/kokkos_ext/ArborX_DetailsKokkosExtStdAlgorithms.hpp:15]: Pulling Kokkos.Core plus all of the Kokkos parallel algorithms to use a simple template function.

**Pull request closed without merging**